tests/connection/test_fsm.py

```diff
@@ -0,0 +1,139 @@
+import pytest
+
+from supa.connection.fsm import LifecycleStateMachine, ProvisionStateMachine, ReservationStateMachine
+from supa.db.model import Reservation
+
+
+def test_reservation_state_machine() -> None:  # noqa: D103
+    reservation = Reservation()
+    rsm = ReservationStateMachine(reservation, state_field="reservation_state")
+    #
```

Comment:
cleanup?

Reply:
@acidjunk What is it that should be cleaned up?

Reply:
Never mind: i thought it was commented code; but it actually explains the flow. (docs :))